qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/test/test_plot_presenter.py

```diff
@@ -241,6 +241,10 @@ def test_on_worker_error_KeyboardInterrupt(self, mock_logger):
         self.presenter._on_worker_error(error_message)
         mock_logger.assert_not_called()
 
+    def test_toggle_legend(self):
+        self.presenter.toggle_legend()
+        self.view.update_figure.assert_called_once()
```

Comment:
I feel the test should be more complete, to make sure that `ax.get_legend().set_visible()` is called once with the correct arguments. 

You could set `self.toolbar.get_show_legend_value = MagicMock(return_value = False)` and check that `set_visible` is called with the argument `False`. Or something similar that checks that the legend itself is called.